### DISCUSSION TITLE
How to write zarr for easy reading back with Xarray

### DISCUSSION THREAD

**Discussion by marstonsward** [pydata/xarray#7554] [Q&A] [answered]
I have a `zarr` store with the following arrays:

```
/
|------->Parentname
|             |____varb3d (1, 24, 1, 721, 1440) float32 
|             |____varb2d (1, 52, 1, 721, 1440) float32
|-------->Metadata
                |-----LEV (31) float32
                |-----fname2 (52) S28
                |-----fname3 (24) S28
                |-----lat (721) float32
                |-----lon (1440) float32
```
But `xarray` cannot read it. Is there a document that explains how to write these `zarr` arrays so that `xarray` can read them?

**Comment by TomNicholas** [comment]
> But xarray cannot read it.

Hi @Marston - can you please read our [guidelines for github discussions](https://github.com/pydata/xarray/discussions/5404)? It's not immediately obvious to me what is going wrong here, and without an error message, copy-pastable example, or an example zarr store that we can access we cannot easily help you further.

> Is there a document that explains how to write these zarr arrays so that xarray can read them?

There is [documentation on using xarray with zarr](https://docs.xarray.dev/en/stable/user-guide/io.html#zarr) - does that help you with your issue?

If you are trying to open both groups at once simultaneously then you would need to use [xarray-datatree](https://github.com/xarray-contrib/datatree), but I'm unclear as to whether that is what you are actually trying to do.

**Reply by keewis** [reply]
there's also the [zarr encoding spec](https://docs.xarray.dev/en/stable/internals/zarr-encoding-spec.html), in case you really need to manually create a zarr store.

**Reply by marstonsward** [reply]
Apologies for the obscure question. This was more a design question. I created a `zarr` store but `xarray` could not read it because it cannot find the dimensions:
```
KeyError: 'Zarr object is missing the attribute `_ARRAY_DIMENSIONS`, which is required for xarray to determine variable dimensions.'
```
But I was creating a group and filling it. It might be easier to append a set of `dataarray` to a store using `to_zarr()`, which would create a `zarr` that can easily be read by `xarray`.

**Reply by TomNicholas** [reply, marked as answer]
If you actually need to create a zarr store manually then you will need to follow the spec, as @keewis linked, but it will probably be much easier to create an xarray.Dataset and use .to_zarr as you say.
